cdc_replication_test.py

```diff
@@ -0,0 +1,193 @@
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+import shutil
+import sys
+import os
+import time
+from tempfile import TemporaryDirectory
+from textwrap import dedent
+
+from cassandra import ConsistencyLevel
+from cassandra.query import SimpleStatement
+
+from sdcm.tester import ClusterTester
+
+def print_file_to_stdout(path: str) -> None:
+    with open(path, "r") as f:
+       shutil.copyfileobj(f, sys.stdout)
+
+# Write a CQL select result to a file.
+def write_cql_result(res, path: str):
+    with open(path, 'w') as f:
+        for r in res:
+            f.write(str(r) + '\n')
+        f.flush()
+        os.fsync(f.fileno())
+
+SCYLLA_MIGRATE_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-migrate"
+REPLICATOR_URL = "https://kbr-scylla.s3-eu-west-1.amazonaws.com/scylla-cdc-replicator-0.0.1-SNAPSHOT-jar-with-dependencies.jar"
+
+def keyspace_def(rf: int) -> str:
+    return f"create keyspace if not exists ks with replication = {{'class': 'SimpleStrategy', 'replication_factor': {rf}}}"
+
+TABLE_DEF_BASE = "create table if not exists ks.tb (pk int, ck int, v text, primary key (pk, ck))"
+TABLE_DEF_MASTER = TABLE_DEF_BASE + " with cdc = {'enabled': true}"
+
+class CDCReplicationTest(ClusterTester):
+    def test_replication(self):
+        print("Adding node to db_cluster... ")
+        node2 = self.db_cluster.add_nodes(count=1, enable_auto_bootstrap=True)[0]
+        print("Waiting for added node to initialize...")
+        self.db_cluster.wait_for_init(node_list=[node2])
+
+        print("Adding node to db_cluster... ")
+        node3 = self.db_cluster.add_nodes(count=1, enable_auto_bootstrap=True)[0]
+        print("Waiting for added node to initialize...")
+        self.db_cluster.wait_for_init(node_list=[node3])
+
+        master_node = self.db_cluster.nodes[0]
+        replica_node = self.cs_db_cluster.nodes[0]
+
+        print('Creating table on master cluster.')
+        with self.cql_connection_patient(node=master_node) as sess:
+            sess.execute(keyspace_def(3))
+            sess.execute(TABLE_DEF_MASTER)
+
+        print('Creating table on replica cluster.')
+        with self.cql_connection_patient(node=replica_node) as sess:
+            sess.execute(keyspace_def(1))
+            sess.execute(TABLE_DEF_BASE)
+
+        print('Waiting 40s for the latest CDC generation to start...')
+        time.sleep(40)
+
+        print("Starting cassandra-stress")
+        stress_thread = self.run_stress_cassandra_thread(stress_cmd=self.params.get('stress_cmd'))
+
+        print('Let C-S run for a while...')
+        time.sleep(10)
+
+        print("Starting nemesis.")
+        self.db_cluster.add_nemesis(nemesis=self.get_nemesis_class(), tester_obj=self)
+        self.db_cluster.start_nemesis()
+
+        loader_node : BaseNode = self.loaders.nodes[0]
+
+        print('Installing tmux on loader node.')
+        res = loader_node.remoter.run(cmd='sudo yum install -y tmux')
```

Comment:
We can add that to loader setup if that's really needed.

Reply:
There are probably other ways to achieve the effect that I'm achieving using tmux, but I know and like tmux so I used it in this test.

So if it's not a problem, it would be nice to have tmux in the setup (it's a small utility so shouldn't be a problem).

Reply:
I like tmux alot too, and there shouldn't be a problem to install it. just move it to the `BaseLoaderSet.node_setup()`, I think you can add it next to `screen` install
 
but for the long run, we usually care about the output of those kind of processes, and in lots of case we take metrics out of it, and search for error on it during runtime.

Reply:
> but for the long run, we usually care about the output of those kind of processes, and in lots of case we take metrics out of it, and search for error on it during runtime.

I wonder what would be the best way to get output "in real time". Spawn a thread and poll for the output?

Reply:
Why didn't you use screen/nohup? 
Anyway, I think we must not install additional tools that are test specific for every run that we have, so let's keep it here.

Reply:
I never used screen/nohup, I always used tmux :)